SharpMUSH.Database.ArangoDB/ArangoDatabase.cs

```diff
@@ -1703,18 +1703,94 @@ public IAsyncEnumerable<SharpAttributeFlag> GetAttributeFlagsAsync(CancellationT
 			$"FOR v in {DatabaseConstants.AttributeFlags:@} RETURN v",
 			cache: true, cancellationToken: ct);
 
-	public ValueTask<bool> ClearAttributeAsync(DBRef dbref, string[] attribute, CancellationToken ct = default)
+	public async ValueTask<bool> ClearAttributeAsync(DBRef dbref, string[] attribute, CancellationToken ct = default)
 	{
-		// Set the contents to empty.
+		// Set the contents to empty, or remove entirely if no children.
+		attribute = attribute.Select(x => x.ToUpper()).ToArray();
+
+		// Get the attribute
+		var attrs = await GetAttributeAsync(dbref, attribute, ct);
+		if (attrs is null) return false;
+
+		var attrArray = await attrs.ToArrayAsync(ct);
+		if (attrArray.Length == 0) return false;
+
+		var targetAttr = attrArray.Last();
+
+		// Check if attribute has children (just need to know if any exist)
+		var children = await arangoDb.Query.ExecuteAsync<string>(handle,
+			$"FOR v IN 1..1 OUTBOUND {targetAttr.Id} GRAPH {DatabaseConstants.GraphAttributes} LIMIT 1 RETURN v._id",
+			cancellationToken: ct);
 
-		throw new NotImplementedException();
+		if (children.Any())
+		{
+			// Has children, just clear the value
+			await arangoDb.Document.UpdateAsync(handle, DatabaseConstants.Attributes,
+				new { Key = targetAttr.Key, Value = string.Empty },
+				mergeObjects: true, cancellationToken: ct);
+		}
+		else
+		{
+			// No children, remove the attribute
+			await arangoDb.Graph.Vertex.RemoveAsync(handle, DatabaseConstants.GraphAttributes,
+				DatabaseConstants.Attributes, targetAttr.Key, cancellationToken: ct);
+		}
+
+		return true;
 	}
 
-	public ValueTask<bool> WipeAttributeAsync(DBRef dbref, string[] attribute, CancellationToken ct = default)
+	public async ValueTask<bool> WipeAttributeAsync(DBRef dbref, string[] attribute, CancellationToken ct = default)
 	{
 		// Wipe a list of attributes. We assume the calling code figured out the permissions part.
+		attribute = attribute.Select(x => x.ToUpper()).ToArray();
+
+		// Get the attribute
+		var attrs = await GetAttributeAsync(dbref, attribute, ct);
+		if (attrs is null) return false;
 
-		throw new NotImplementedException();
+		var attrArray = await attrs.ToArrayAsync(ct);
```

Comment:
Don't immediately change to an Array. Stay in AsyncEnumerable mode as long as possible to avoid too many memory allocations.